### PR TITLE
Add support for layer-specific perturbations and save intermediate Tyche values

### PR DESCRIPTION
**Pull request by dipikakhullar**
Saving internals during perturbation:
1. mults — perturbation multipliers
2. deltas — parameter changes
3. props — possibly KL divergence or other proportional metrics

Support for layer-specific perturbations:
1.  added logic to restrict perturbation to e.g. layer 15, using a layer mask or filter string when computing implicit vectors.

### CODE REVIEW

File: src/tyche/estimator.py
Comment (davidoj) [inline src/tyche/estimator.py:1]: 214-232: It looks like you're computing logits_p twice (once going through the val_data all at once, once batched). I think we should retain the original batched logic, you can always set batch size = val_data size.

File: src/tyche/estimator.py [hunk 1]
```diff
@@ -221,11 +235,18 @@ def apply_fn(params, x):
...
+            # if self.config.cache_mode:
+            #     self.logits_p = torch.cat(logits_p_list, dim=0)  # Raw unperturbed logits (same shape as val_data)
+
         else:
             self.probs_p = None
         
         def kl_fn(a, b, mults=None):
+            # For capturing final logits (you'll need to add flags to indicate when to save them)
+            final_logits_original = []
+            final_logits_perturbed = []
```
Comment (davidoj) [inline src/tyche/estimator.py]: what's the difference between latest_original_logits and final_logits_original?

File: src/tyche/estimator.py [hunk 2]
```diff
@@ -136,7 +142,9 @@ def run(self) -> VolumeResult:
             iters=self.config.iters,
             allow_unconverged=self.config.allow_unconverged,
             init_mult=self.config.init_mult,
-            rtol=self.config.rtol,
+            rtol=self.config.rtol
+            # perturbation_mask=self.config.perturbation_mask
```
Comment (davidoj) [inline src/tyche/estimator.py]: Delete lines we're not using

**Comment by davidoj** [comment]
It looks like the main change here is adding the module filter, is that right? It looks like you also added the code to save logits and then removed it again. Saving logits seems useful, at least as an option, but if it's not fully implemented better to delete the related code than comment it I think.
